iree/compiler/Conversion/CodegenUtils/FunctionUtils.cpp

```diff
@@ -42,12 +44,42 @@ IREE::HAL::ExecutableEntryPointOp getEntryPoint(FuncOp funcOp) {
   return nullptr;
 }
 
-Value getViewSource(Value view) {
-  while (auto viewOp = view.getDefiningOp<ViewLikeOpInterface>()) {
-    view = viewOp.getViewSource();
+/// Walk up the defs of the view, to get the untiled value. Either walks up
+/// `ViewOpInterface` op-chains or the `subtensor` op-chains.
+static Value getViewSource(Value view) {
+  while (true) {
+    Operation *definingOp = view.getDefiningOp();
+    if (!definingOp) break;
+    if (auto viewOp = view.getDefiningOp<ViewLikeOpInterface>()) {
+      view = viewOp.getViewSource();
+      continue;
+    }
+    if (auto subTensorOp = view.getDefiningOp<SubTensorOp>()) {
+      view = subTensorOp.source();
+      continue;
+    }
+    if (auto dispatchTensorLoadOp =
+            view.getDefiningOp<IREE::Flow::DispatchTensorLoadOp>()) {
+      view = dispatchTensorLoadOp.source();
```

Comment:
I thought this got updated to support ViewLikeOpInterface? or was I thinking of something else?
(it'd be nice if we didn't have a dep on flow here as it's a layering violation)

Reply:
I think `ViewLikeOpInterface` works only memref types. You are probably thinking of `OffsetsSizesAndStridesInterface`